File: ckgame-java/src/main/java/com/ckgame/military/UnitType.java
package com.ckgame.military;

/**
 * 单位类型。
 */
public enum UnitType {
    LEVIES,
    LIGHT_INFANTRY,
    HEAVY_INFANTRY,
    PIKEMEN,
    ARCHERS,
    LIGHT_CAVALRY,
    HEAVY_CAVALRY,
    KNIGHTS;

    /** 伤害（攻击力）。 */
    public double damage() {
        return switch (this) {
            case LEVIES -> 5.0;
            case LIGHT_INFANTRY -> 8.0;
            case HEAVY_INFANTRY -> 14.0;
            case PIKEMEN -> 12.0;
            case ARCHERS -> 10.0;
            case LIGHT_CAVALRY -> 12.0;
            case HEAVY_CAVALRY -> 20.0;
            case KNIGHTS -> 28.0;
        };
    }

    /** 坚韧（防御/生命值）。 */
    public double toughness() {
        return switch (this) {
            case LEVIES -> 4.0;
            case LIGHT_INFANTRY -> 6.0;
            case HEAVY_INFANTRY -> 12.0;
            case PIKEMEN -> 14.0;
            case ARCHERS -> 5.0;
            case LIGHT_CAVALRY -> 8.0;
            case HEAVY_CAVALRY -> 14.0;
            case KNIGHTS -> 18.0;
        };
    }

    /** 每人每月维护费。 */
    public double maintenance() {
        return switch (this) {
            case LEVIES -> 0.05;
            case LIGHT_INFANTRY -> 0.1;
            case HEAVY_INFANTRY -> 0.25;
            case PIKEMEN -> 0.2;
            case ARCHERS -> 0.15;
            case LIGHT_CAVALRY -> 0.3;
            case HEAVY_CAVALRY -> 0.5;
            case KNIGHTS -> 0.8;
        };
    }
}

File: ckgame-ts/src/military/UnitType.ts
/**
 * Unit types with combat stats.
 */
export enum UnitType {
  LEVIES = 'LEVIES',
  LIGHT_INFANTRY = 'LIGHT_INFANTRY',
  HEAVY_INFANTRY = 'HEAVY_INFANTRY',
  PIKEMEN = 'PIKEMEN',
  ARCHERS = 'ARCHERS',
  LIGHT_CAVALRY = 'LIGHT_CAVALRY',
  HEAVY_CAVALRY = 'HEAVY_CAVALRY',
  KNIGHTS = 'KNIGHTS',
}

const unitDamage: Record<UnitType, number> = {
  [UnitType.LEVIES]: 5.0,
  [UnitType.LIGHT_INFANTRY]: 8.0,
  [UnitType.HEAVY_INFANTRY]: 14.0,
  [UnitType.PIKEMEN]: 12.0,
  [UnitType.ARCHERS]: 10.0,
  [UnitType.LIGHT_CAVALRY]: 12.0,
  [UnitType.HEAVY_CAVALRY]: 20.0,
  [UnitType.KNIGHTS]: 28.0,
};

const unitToughness: Record<UnitType, number> = {
  [UnitType.LEVIES]: 4.0,
  [UnitType.LIGHT_INFANTRY]: 6.0,
  [UnitType.HEAVY_INFANTRY]: 12.0,
  [UnitType.PIKEMEN]: 14.0,
  [UnitType.ARCHERS]: 5.0,
  [UnitType.LIGHT_CAVALRY]: 8.0,
  [UnitType.HEAVY_CAVALRY]: 14.0,
  [UnitType.KNIGHTS]: 18.0,
};

const unitMaintenance: Record<UnitType, number> = {
  [UnitType.LEVIES]: 0.05,
  [UnitType.LIGHT_INFANTRY]: 0.1,
  [UnitType.HEAVY_INFANTRY]: 0.25,
  [UnitType.PIKEMEN]: 0.2,
  [UnitType.ARCHERS]: 0.15,
  [UnitType.LIGHT_CAVALRY]: 0.3,
  [UnitType.HEAVY_CAVALRY]: 0.5,
  [UnitType.KNIGHTS]: 0.8,
};

export function unitDamageValue(u: UnitType): number {
  return unitDamage[u] ?? 1.0;
}

export function unitToughnessValue(u: UnitType): number {
  return unitToughness[u] ?? 1.0;
}

export function unitMaintenanceValue(u: UnitType): number {
  return unitMaintenance[u] ?? 0.1;
}
